tests/common/util.rs

```diff
@@ -1191,12 +1243,525 @@ impl UCommand {
+                let stderr = self.raw.stderr.as_mut().unwrap();
+                stderr.read_to_end(&mut buffer).unwrap();
+                buffer
+            }
+            None => vec![],
+        }
+    }
+
+    /// Returns all output from start of the child processes' stderr until now, if we captured the
+    /// output or else returns the output of the child process that can currently be read. See also
+    /// [`UChild::stdout_all_bytes`] for additional information.
+    pub fn stderr_all_bytes(&mut self) -> Vec<u8> {
+        match self.captured_stderr.as_mut() {
+            Some(output) => output.output_all_bytes(),
+            None => self.stderr_bytes(),
+        }
+    }
+
+    /// Reads, consumes and returns the exact amount of bytes from stderr. If stderr is redirect to
+    /// stdout with [`UCommand::stderr_to_stdout`] then always zero bytes are returned.
+    ///
+    /// # Important
+    /// This method blocks indefinitely if the `size` amount of bytes cannot be read.
+    pub fn stderr_exact_bytes(&mut self, size: usize) -> Vec<u8> {
+        match self.captured_stderr.as_mut() {
+            Some(output) => output.output_exact_bytes(size),
+            None if self.raw.stderr.is_some() => {
+                let stderr = self.raw.stderr.as_mut().unwrap();
+                let mut buffer = vec![0; size];
+                stderr.read_exact(&mut buffer).unwrap();
+                buffer
+            }
+            None => vec![],
+        }
+    }
+
+    /// Returns true if the child process is still alive and false otherwise.
+    ///
+    /// # Panics
+    /// If an error occurred during [`Child::try_wait`]
+    pub fn try_alive(&mut self) -> bool {
+        // TODO: return false on error of try_wait ?
+        self.raw.try_wait().unwrap().is_none()
+    }
+
+    /// Returns a [`UChildAssertion`] on which helper functions with assertions can be called.
+    pub fn make_assertion(&mut self) -> UChildAssertion {
+        UChildAssertion::new(self)
+    }
+
+    pub fn make_assertion_with_delay(&mut self, millis: u64) -> UChildAssertion {
+        sleep(Duration::from_millis(millis));
+        self.make_assertion()
+    }
+
+    /// Pipe data into [`Child`] stdin in a separate thread to avoid deadlocks. Note, that
+    /// [`UCommand::set_stdin`] must be used together with [`Stdio::piped`] or else this method
+    /// doesn't work as expected. `Stdio::piped` is the current default when using
+    /// [`UCommand::run_no_wait`]) without calling `set_stdin`. This method returns a [`JoinHandle`]
+    /// of the thread in which the writing to the child processes stdin is running.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the [`UCommand::set_stdin`] is called with setting [`Stdio::null`] or
+    /// the child's stdin cannot be taken out.
+    ///
+    /// [`JoinHandle`]: std::thread::JoinHandle
+    pub fn pipe_in<T: Into<Vec<u8>>>(&mut self, content: T) -> JoinHandle<io::Result<()>> {
```

Comment:
I feel like I would always want the thread to finish writing, so the behaviour of `pipe_in_and_wait`.

Reply:
Please have a look at how it is implemented now. Usually you don't have to care about the join, because it's done automatically in the wait methods.